confirms.go

```diff
@@ -29,11 +33,8 @@ func (c *confirms) Listen(l chan Confirmation) {
 
 // publish increments the publishing counter
 func (c *confirms) Publish() uint64 {
-	c.m.Lock()
-	defer c.m.Unlock()
-
-	c.published++
-	return c.published
+	c.published.Store(c.published.Load().(uint64) + 1)
```

Comment:
This seems racy. What prevents another process from altering c.published between Load and store?

Instead, IMHO, just using atomic.AddUint64(&c.published) to increment c.published and removing the locking should be all that's necessary to accomplish lock-less incrementing.

Reply:
I didn't know that its possible to do `atomic.AddUint64(&c.published)`. I'll update my PR. 
Thanks for the insight!